github-git/0-first-contribution.md

```diff
@@ -0,0 +1,188 @@
+
+**Fork the pyOpenSci practice GitHub repository**
+*******
+* Navigate to the [pyOpenSci example repo](https://github.com/pyOpenSci/pyos-demo-package-contribute).
+* Fork the repository.
+
+*Remember that a fork is a copy of a repository that is owned by someone else or an organization that lives in your GitHub account.*
+:::
+
+********
+
+### Step 4: Edit and commit your changes
+
+Once you've successfully forked the repo, it's time to edit the file you want to fix. In this lesson, you are editing documentation files in the GitHub interface. This means that you don't need to set up a local development environment.
+
+:::{button-link} edit-commit-files
+:color: primary
+:shadow:
+
+<i class="fa-brands fa-leanpub" style="color: #6ec9c3;"></i> Learn how to  edit & commit files on GitHub
+:::
+
+:::{admonition} Activity: Make a change to a file and commit it to your fork
+
+1. Navigate to your fork on GitHub.com
+2. In the GitHub interface, click on the file that you proposed to modify or fix [in the identify issue lesson](identify-issue)
+3. Click on the edit <kbd><i class="fa-solid fa-pencil"></i></kbd> button in the GitHub interface.
+4. Make the edits to the file that you proposed in your issue.
+5. Hit the <kbd>commit</kbd> button to save your edits
+6. Add a descriptive commit message that describes the change that you made
+
+Commit message examples:
+> fix: fixed numerous typos in the filename.py file
+>
+> fix: updated the code to align with PEP 8 syntax
+>
+> fix: fixed typo in docstring text
+
+:::
+
+******
+
+### Step 5: Submit a pull request
+
+Once your edits are [committed to git version control](edit-commit-files), open a <kbd>Pull Request</kbd> to the parent repository.
+
+:::{button-link} pull-request.html
+:color: primary
+:shadow:
+
+<i class="fa-brands fa-leanpub" style="color: #6ec9c3;"></i> Learn how to create a GitHub pull request
+:::
+
+:::{admonition} Activity: Submit a PR to the pyOpenSci parent repository
+Create a <kbd><i class="fa-solid fa-code-pull-request"></i> Pull Request</kbd> to the [pyOpenSci parent repository](https://github.com/pyOpenSci/pyos-demo-package-contribute), from the branch on your fork that you worked on in the previous lesson.
+:::
+
+Once your PR is open, it's time to sit back and wait for the maintainers, collaborators, or project owners to review/comment on your PR. Be patient; this step can take time as people are busy and often donate their time to this effort!
+
+********
+
+## Many people's first contributions are to documentation 
+
+In this lesson, you make your first open source contribution by updating documentation (including docstrings) rather than code. Many people start with small fixes like typos, which are simpler to contribute as you don't always need a development environment but are still valuable. Developers often focus on code, leaving less time for writing and editing documentation. So your small contributions can have a large impact.
```

Comment:
from @ucodery - some projects do NOT welcome small pr's. i need to know more about that to write about it! i had no idea that was a thing and I've never encountered it. Jeremiah, any feedback is welcome!!